corehq/apps/data_interfaces/models.py

```diff
@@ -76,25 +96,44 @@ def _rule_matches_case(self, case, now):
         if case.type != self.case_type:
             return False
 
-        if (case.server_modified_on >
-                (now - timedelta(days=self.server_modified_boundary))):
+        if self.filter_on_server_modified and \
+                (case.server_modified_on > (now - timedelta(days=self.server_modified_boundary))):
             return False
 
         return all([criterion.matches(case, now)
                    for criterion in self.automaticupdaterulecriteria_set.all()])
 
     def apply_actions(self, case):
-        properties = {}
+        cases_to_update = defaultdict(dict)
         close = False
 
+        def _get_case_property_value(current_case, name):
+            return current_case.resolve_case_property(name)[0].value
+
+        def _add_update_property(name, value, current_case):
+            while name.startswith('parent/'):
+                name = name[7:]
+                current_case = current_case.parent
```

Comment:
we'll want to instead use `current_case.get_parent(identifier=DEFAULT_PARENT_IDENTIFIER)` (which returns a list that can be empty) and take the first one. `current_case.parent` assumes the relationship is normal parent/child without allowing for other relationships, like extension case relationships